src/aead/quic.rs

```diff
@@ -76,6 +112,10 @@ impl Algorithm {
     /// The length of the key.
     #[inline(always)]
     pub fn key_len(&self) -> usize { self.key_len }
+
```

Comment:
I've added this, but it's not clear to me what the use case for it is. QUIC selects the header protection algorithm based on the AEAD algorithm, so the AEAD algorithm is always known first.

Reply:
> I've added this, but it's not clear to me what the use case for it is. QUIC selects the header protection algorithm based on the AEAD algorithm, so the AEAD algorithm is always known first.

That's true, but with this API you can instead map the QUIC algorithm identifier to the protection key algorithm, and then get the AEAD algorithm, since there is a 1:1 correspondence (as of now). That said, it's not a big savings of complexity so if you think it is too confusing (because it seems to be the opposite of how the spec says they should be mapped), I'm OK with not adding it.

Reply:
> you can instead map the QUIC algorithm identifier to the protection key algorithm, and then get the AEAD algorithm

I'm not following something here. QUIC starts with the negotiated TLS cipher suite (using a hardcoded AEAD/digest before negotiation completes), looks up an AEAD from that, and looks up a header protection algorithm based on that. You never start out knowing what header protection algorithm to use.

Reply:
Did you mean to suggest looking up the header protection algorithm based on the TLS ciphersuite? rustls already exposes the AEAD algorithm from the cipher suite directly, so it's not clear to me how that would be a complexity savings at all.

Reply:
OK, let's just drop it then. I think some people may be building QUIC implementations that aren't based on Rustls, and it might be useful for them. But if so, they can submit a PR to add it themselves. Until then let's leave it out.

Reply:
Done.